tests/emit_test.toit

```diff
@@ -65,7 +64,25 @@ main:
     diff /Duration := duration - expected
     // Should never trigger earlier.
     expect diff >= Duration.ZERO
+
     // We allow up to half a dot-duration of being too late.
-    expect diff < (dot_duration / 2)
+    // Don't use `expect` as it exits.
+    if not diff < (dot_duration / 2):
+      throw "Dot was too long"
 
     last_time = this_time
+
+main:
+  // We start with a duration that is quite aggressive.
+  // The test might fail. If it does we increment the time.
+  dot_duration := Duration --ms=1
+  10.repeat:
+    catch:
```

Comment:
This irks me a bit (catch all) and I think I'd prefer passing a fail block to `run_test` and never let that use `expect` and exceptions.

Reply:
Passing in a `--too-slow` block which is called.